[Android] Top bar item colors - fix

kubaflo opened the pull request:
### Description
Android docs: https://developer.android.com/reference/com/google/android/material/tabs/TabLayout#setTabIconTint(android.content.res.ColorStateList)

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/26905

|Before|After|
|--|--|
<video src="https://github.com/user-attachments/assets/0651a427-67a8-40c9-95cc-a37c80970f53" width="300px"/>|<video src="https://github.com/user-attachments/assets/a7971630-7f33-4598-a34b-d0b2972a24c8" width="300px"/>|

@jsuarezruiz what do you think?

jfversluis commented:
/azp run

Tamilarasan-Paranthaman commented:
@kubaflo, I believe removing SetIconColorFilter is not appropriate, as it could cause issues in certain scenarios involving FontImageSource color.
 
Recently, I fixed the FontImageSource color issue on the TabbedPage icon. The PR can be found here: [[iOS] [Android] Fix for the FontImageSource color is not applied properly to the Tab Icon by Tamilarasan-Paranthaman · Pull Request #26757 · dotnet/maui](https://github.com/dotnet/maui/pull/26757)
 
For instance, if there are two ContentPage instances as children of a TabbedPage, setting SelectedTabColor to red while applying different FontImageSource colors to each page may not work as expected.
 
The SetIconColorFilter method handles the Icon.InvalidateSelf call, ensuring that the icon is updated correctly.

kubaflo commented:
Hi @Tamilarasan-Paranthaman this pr actually fixes the issue in your PR. And there's no need for `SetIconColorFilter`


https://github.com/user-attachments/assets/4c700673-794a-42bb-bdd7-bb39fd06df84

rmarinho commented:
Can we make sure adding a menu item and disable and enable it with have the correct colors?

Tamilarasan-Paranthaman commented:
> Hi @Tamilarasan-Paranthaman this pr actually fixes the issue in your PR. And there's no need for `SetIconColorFilter`
> 
>  Screen.Recording.2025-01-06.at.18.09.05.mov

@kubaflo, can you please fully utilize my test sample? I have added the SelectedTabColor and UnSelectedTabColor, but no FontImageSource color is specified for the first page. Only the second and third pages have the FontImageSource color defined.
 
With your fix, the FontImageSource color is not applied to the second and third pages.
 
 
![undefined](https://github.com/user-attachments/assets/ae85184b-2611-45a5-8279-65bd3a0bc8ce)

 
I have handled the FontImageSource color for different scenarios by using combinations of SelectedTabColor, UnSelectedTabColor, and the FontImageSource color in my PR. It works properly.

jsuarezruiz commented:
/azp run

kubaflo committed:
[Android] Top bar item colors - fix

jsuarezruiz commented:
/azp run

jsuarezruiz commented:
/rebase

jsuarezruiz commented:
/azp run

PureWeen reviewed (changes requested):
https://github.com/dotnet/maui/pull/26964#issuecomment-2579282600

kubaflo committed:
Avoid tab icon tint for FontImageSource on Android

Prevents applying TabIconTint when the tab icon is a FontImageSource, as it manages its own color. This ensures correct icon coloring for font-based icons in TabbedPageManager.

IX-OmkarP commented:
@jfversluis @kubaflo  Thank you it is working
